lib/segment/src/index/hnsw_index/search_context.rs

```diff
@@ -24,18 +22,17 @@ impl SearchContext {
     }
 
     pub fn lower_bound(&self) -> ScoreType {
-        match self.nearest.top() {
-            None => ScoreType::min_value(),
-            Some(worst_of_the_best) => worst_of_the_best.score,
-        }
+        self.nearest.threshold()
     }
 
     /// Updates search context with new scored point.
     /// If it is closer than existing - also add it to candidates for further search
     pub fn process_candidate(&mut self, score_point: ScoredPointOffset) {
         let was_added = match self.nearest.push(score_point) {
             None => true,
-            Some(removed) => removed.idx != score_point.idx,
+            Some(removed) => removed
+                .iter()
+                .any(|removed_point| removed_point.idx == score_point.idx),
```

Comment:
I think it would be faster to not check this and just let it go through next iteration so that:
```
if candidate.score < searcher.lower_bound() {
    break;
}
```
takes care of it. Wdyt?

Reply:
what's a `candidate`?

Reply:
Point whose links are to be explored.